tutorial/source/gmm.ipynb

```diff
@@ -593,6 +594,113 @@
MHLIzc8jKzC75c8lXRnbJn0ted/ZYTmYO+XnF3at96/nQe2Q3Og/pgKNT8Yz3xAMn+W3GCjatT6hwUzwHRxMd+kTR++YuJZ2zC/IL2PDrFpZ+F8upo5ZPvyorWJzbEjYypilRMeHUDw246PsnDicXT3Fau4uE7Uds0km8soJFr1u7MfTh/qSdzOCFG4ub4ZXmXxMsrrnmGgYPHswDDzxw0fGpU6fy448/8vPPP1tecTWiYCEiIlJz2TtYQPEWtI98MBonF0fMZi662c8+k0NOZm7JTX92RtbF37/gKyczh6yMHPJybNOg7kIetd3pfnMXrr6xC67uxdOFUk9nsey79VZNZTIYDETFNKXPLV1oFHF+J6nNq+NZMnMdh3YdL//FLrh5dnRyoGnbkJIw4enjXvI9s9nMvq2H2L5uN9vW7rLpdrjnVEawcPN05YU5j+Pq7nxRM7zS/GuChY+PD2vXriUiIuKi4/Hx8cTExJCcXP4t2KojBQsREZGaqzKCBYCjsyMmk5Hc7LxL/sa8qm8QXdyc6Xp9ND1HdMMrwBsonsq0au5GVs7ZyJn0it9Qh7VsSK+bO9OiU5OSY3s2H2DJt+vZ+fu+y57v7ulCZKcmRMaEE9Gh8UVbwuZm57Hz931sW7uLHbF7yLKizvKojGAxbPwgut/U+R/N8EpT1Z+bC9l1u9nc3FwKCgr+cTw/P5/sKtgDWERERKSy5efmU/okluolJyuXJV+tYsWsdXQaFlMylan/bVfR6+YurPtpE8tnbyAlKc3ia+/deoi9Ww9RN9i/ZCepJq2DadI6mGP7k1jy7Xo2Lf/rop2kAhv5EdWlKZFdmhJyrnfGWakn09m6Zhfb1+1i7+aDVb4lrC35B/ly1bBOAMx592ebTN+qjix+YtGjRw8iIyOZMmXKRccffPBBtm7dyurVq21aYGXTEwsREZGaq7KeWJRXdfrNs8nLC4PBQKtu4fQZEUNQ07oAFBUWEbdsB0tmruP4gZMVvr63vyc9hkXTZVCbkp2kTp9IY9XcP/DydSeyS1P86tU+f4LZzOHdx0vWSxzdW3Xbrtr7icWlmuGVpjp9buw6FWrt2rX07t2bDh060KtXLwCWLl3Kxo0bWbx4Md26dat45dWAgoWIiEjNpWBRtr/vCtW0bQh9RnQhvG1IybHt63bz28x1JOw4UuFx3DxciBncju43dCzZSeqcgvwCdm86ULwl7Np40k79c6vVqmDPYBHaOpjxU++hqMjMqyOnlKtvRXX63Nh9u9nNmzczadIktmzZgqurKy1btuTpp5+mSZMmlz+5mlOwEBERqbkULMpW1nazDcPr0vuWLrTu1gwMxVvV7ojdw+cvzilz16LycHA0Ed2vFe16tSD5eCrb1u1mV1zC+YXj1ei9sVewqN+kLne8MJy6jeuwdt5GZk6cV67zqtPnRn0srKBgISIiUnMpWJTtcn0s/Bv40PumznTs1xKTg4l9Ww8x9elvbdMMrzTV6L2xdbBw967FoPv60OXa9hiNBrLSs/nfzW+TkZJZrvOr0+fG7sGisLCQefPmsXPnTgBatGjBkCFDMFWz/5krQsFCRESk5lKwKFt5G+Q1alaPB1+/FZdazuzfdpgPn5ppn3BRjd4bWwULo8nIVcM6MWB0r5Ku4JuWbGPee7+Skpha7utUp8+NXYPF3r17GThwIEeOHCE8PByAXbt2ERQUxMKFCwkNDa145dWAgoWIiEjNpWBRNks6bzcMr8uDr9+Kq7sLCTuO8MGT35CbZeNwUY3eG1sEi4hOTRg2bhABwX4AHNmdyPeTF7D3zwMWX6s6fW7sGiwGDBiA2Wzm66+/Lum+nZyczMiRIzEajSxcuLDilVcDChYiIiI1l4JF2SwJFgBBTevy4OsjcPNw5cBfR/ngyW/IOZNru4Kq0XtjTbDwD/Ll+rEDiOrWDIDM1CwWTF3Muvl/VHhb2er0ubFrsKhVqxaxsbFERUVddHzLli3ExMSQmVm+uWPVlYKFiIhIzaVgUTZLgwVAgyaBPPTGrbh5uHIw/hgfPPE12Zk2ChfV6L2pSLBwcXOm/1096HFzDCYHI0WFZlbOXsfPny4jOzPHqnqq0+fGkmBhtPTizs7OZGT8c2uwzMxMnJycLL2ciIiIiFRTR/YkMmXCV2RlZJesvXB1d67qsqqUwWCg06B2PPf9o/Qe2Q2Tg5G/1u/m5RHv8MPbP1sdKmoyi4PFoEGDuPfee9mwYQNmsxmz2UxsbCxjxoxhyJAh9qhRRERERKrI0X1JTJnwFWfSs2kYXo+H3hhZsjD53yYkqiGPf/4AI5+9Hg+fWpw4lMyHj87gg/HTSTpY8eaCVwqLp0KlpqZy++23s2DBAhwdi6cKFRQUMGTIEL744gu8KvCYrTrRVCgREZGaS1OhylaRqVAXqhtSh4ffHIm7txtH9iTy3mNfkZVhxW/nq9F7c7mpUN7+nlz3UH/a92sFQG5WHj9/upQV362nsMD2P0d1+txUSh+LPXv2EB8fD0BERARhYWEVuUy1o2AhIiJScylYlM3aYAEQ2MiPh98ahUftWsVPMh79iqz0Ci58rkbvTVnBwtHJgV63dqPv7Vfj5OKI2QyxC+L48cNFZJw+Y7d6qtPnRg3yrKBgISIiUnMpWJTNFsECLg4Xx/afnyZlsWr03pQWLFp3b8HQsdfgW682APu3HmL2Wws4HH/M7vVUp8+NzYPFhAkTyj34W2+9Ve7XVkcKFiIiIjWXgkXZbBUsAAKCfBk7eRQePu4cSzjBexO+IjMty7KLVKP35sJgUS8skBsnDKJJ2xAAUk+kM3fKL8T9trXS6qlOnxtLgoVDeS74559/lmtgg8FQrteJiIiISM2VdDiZd8bPYOxbt1EvpA5jJ4/i3QlfkZlqv+lB9lbLy43B9/Why3UdMBoN5OcVsuTLlfz25SrycvKrurwaQVOh/kZPLERERGouPbEomy2fWJzj38CHsZNH4eXrQeKBk5aFi2ry3hhNRroOaMm
```

Comment:
This is nice to see! I have found seaborn's density plots to be quite useful for such visualizations out of the box `sns.kdeplot(X.numpy(), Y.numpy())`, if you don't mind the extra dependency. It will likely be more smoothed out.